src/main/java/io/vertx/codegen/CodeGenProcessor.java

```diff
@@ -273,11 +275,32 @@ public ModelProcessing(Model model, CodeGenerator generator) {
   }
 
   private static class GeneratedFile extends ArrayList<ModelProcessing> {
-    final File file;
+    final String uri;
     final Map<String, Object> session = new HashMap<>();
-    public GeneratedFile(File file) {
+    public GeneratedFile(String uri) {
       super();
-      this.file = file;
+      this.uri = uri;
+    }
+
+    void writeTo(Writer writer) throws IOException {
+      Collections.sort(this, (o1, o2) ->
+        o1.model.getElement().getSimpleName().toString().compareTo(
+          o2.model.getElement().getSimpleName().toString()));
+      for (int i = 0; i < size(); i++) {
+        ModelProcessing processing = get(i);
+        Map<String, Object> vars = new HashMap<>();
+        vars.put("incrementalIndex", i);
+        vars.put("incrementalSize", size());
+        vars.put("session", session);
+        try {
+          String part = processing.generator.transformTemplate.render(processing.model, vars);
+          writer.append(part);
+        } catch (GenException e) {
+          throw e;
```

Comment:
it is caught because of the general Exception catch just after that setup an model element for error reporting